src/ray/core_worker/task_manager.cc

```diff
@@ -1030,6 +1034,18 @@ bool TaskManager::RetryTaskIfPossible(const TaskID &task_id,
   }
 }
 
+void TaskManager::RetryTask(TaskEntry *task_entry,
+                            bool object_recovery,
+                            uint32_t delay_ms) {
+  RAY_CHECK(task_entry != nullptr);
+  SetTaskStatus(*task_entry,
```

Comment:
The only difference is that `SetTaskStatus` is called with `mu_` in one place, but not here. I think that’s fine, since not every call to `SetTaskStatus` is guarded by `mu_`.

https://github.com/ray-project/ray/blob/4182a54847b5ecb78d2c0883e12928a439413164/src/ray/core_worker/task_manager.cc#L890-L898